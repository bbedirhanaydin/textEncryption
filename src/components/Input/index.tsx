import React from 'react';
import {View, TextInput, StyleSheet, Dimensions} from 'react-native';

const Input = ({placeholder, onChangeText}: any) => {
  return (
    <View style={styles.container}>
      <TextInput
        testID="input-test"
        editable
        multiline
        style={styles.input_container}
        placeholder={placeholder}
        onChangeText={onChangeText}
        textAlignVertical="top"
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    margin: 10,
  },
  input_container: {
    width: Dimensions.get('window').width / 1.1,
    height: Dimensions.get('window').height / 1.4,
    borderWidth: 1,
    margin: 5,
    padding: 10,
    borderRadius: 5,
    borderColor: '#bdbdbd',
    fontSize: 16,
    color: '#000000',
  },
  label: {
    fontSize: 18,
    fontWeight: 'bold',
    marginLeft: 3,
    alignSelf: 'center',
  },
});

export default Input;
