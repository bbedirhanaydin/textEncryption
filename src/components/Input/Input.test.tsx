import React from 'react';
import {render, fireEvent, screen} from '@testing-library/react-native';
import Input from '.';

test('should match with snapshot input', () => {
  const comp = render(<Input />);
  expect(comp).toMatchSnapshot();
});

test('should control onChangeText', () => {
  const mockFunction = jest.fn();
  const comp = render(<Input onChangeText={mockFunction} />);

  const inputChange = comp.getByTestId('input-test');
  fireEvent(inputChange, 'onChange');

  expect(mockFunction).toBeCalled;
});
