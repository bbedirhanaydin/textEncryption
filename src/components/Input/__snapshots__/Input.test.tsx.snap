// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot input 1`] = `
<View
  style={
    {
      "margin": 10,
    }
  }
>
  <TextInput
    editable={true}
    multiline={true}
    style={
      {
        "borderColor": "#bdbdbd",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#000000",
        "fontSize": 16,
        "height": 952.8571428571429,
        "margin": 5,
        "padding": 10,
        "width": 681.8181818181818,
      }
    }
    testID="input-test"
    textAlignVertical="top"
  />
</View>
`;
